qemu/tests/qnum_change_reboot.py

```diff
@@ -0,0 +1,232 @@
+                              t_option,
+                              params.get_numeric("netperf_para_sessions"),
+                              params.get("netperf_cmd_prefix", ""),
+                              package_sizes=params.get("netperf_sizes"))
+            if utils_misc.wait_for(n_client.is_netperf_running, 10, 0, 1,
+                                   "Wait netperf test start"):
+                logging.info("Netperf test start successfully.")
+            else:
+                test.error("Can not start netperf client.")
+            utils_misc.wait_for(
+                lambda: not n_client.is_netperf_running(),
+                netperf_test_duration, 0, 5,
+                "Wait netperf test finish %ss" % netperf_test_duration)
+            time.sleep(5)
+        finally:
+            n_server.stop()
+
+    login_timeout = params.get_numeric("login_timeout", 360)
+    bg_stress_test = params.get("run_bgstress")
+    bg_stress_run_flag = params.get("bg_stress_run_flag")
+    vm = env.get_vm(params["main_vm"])
+    vm.verify_alive()
+    vm.wait_for_login(timeout=login_timeout)
+    session_serial = vm.wait_for_serial_login(timeout=login_timeout)
+    enable_multi_queues(vm)
+    ping_lost_ratio = params.get_numeric("background_ping_package_lost_ratio", 5)
+    ping_time = params.get_numeric("background_ping_time", 60)
+    guest_ip = vm.get_address()
+    required_reboot = True
+    try:
+        while ping_test:
+            session_serial = vm.wait_for_serial_login(timeout=login_timeout)
+            error_context.context("Run test %s background" % bg_stress_test,
+                                  logging.info)
+            wait_time = params.get_numeric("wait_bg_time", 60)
+            env[bg_stress_run_flag] = False
+            stress_thread = utils_misc.InterruptedThread(
+                netperf_test, (), {})
+            stress_thread.start()
+            if bg_stress_run_flag:
+                utils_misc.wait_for(lambda: env.get(bg_stress_run_flag),
+                                    wait_time, 0, 5,
+                                    "Wait %s start background" % bg_stress_test)
+
+            # ping test
+            error_context.context("Ping guest from host", logging.info)
+            args = (guest_ip, ping_time, ping_lost_ratio)
+            bg_test = utils_misc.InterruptedThread(ping_test, args)
+            bg_test.start()
+
+            error_context.context("Change queues number repeatedly",
+                                  logging.info)
+            repeat_counts = params.get_numeric("repeat_counts")
+            for nic in iter(vm.virtnet):
+                queues = int(nic.queues)
+                if queues == 1:
+                    logging.info("Nic with single queue, skip and continue")
+                    continue
+                ifname = utils_net.get_linux_ifname(session_serial, nic.mac)
+                change_list = params.get("change_list").split(",")
```

Comment:
What is 'change list'? It's hard to get its meaning without reading other code.

Reply:
'change_list' is defined in qnum_change_reboot.cfg.